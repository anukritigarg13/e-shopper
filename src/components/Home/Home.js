import React from 'react';
import PropTypes from 'prop-types';
import ProductList from '../ProductList/ProductList';
import './Home.css';

const Home = ({
  category, products, addItemHandler, removeItemHandler,
}) => (
  
);
const productsShape = {
  id: PropTypes.number.isRequired,
  imgAlt: PropTypes.string.isRequired,
  imgSrc: PropTypes.string.isRequired,
  companyName: PropTypes.string.isRequired,
  itemName: PropTypes.string.isRequired,
  unitPrice: PropTypes.number.isRequired,
  unitQuantity: PropTypes.string.isRequired,
  itemCount: PropTypes.number.isRequired,
  category: PropTypes.string.isRequired,
};
Home.propTypes = {
  category: PropTypes.string.isRequired,
  addItemHandler: PropTypes.func.isRequired,
  removeItemHandler: PropTypes.func.isRequired,
  products: PropTypes.arrayOf(PropTypes.shape(productsShape)).isRequired,
};
export default Home;
