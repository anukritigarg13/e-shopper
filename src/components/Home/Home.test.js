/* eslint-disable react/jsx-props-no-spreading */
import { render, screen } from '@testing-library/react';
import React from 'react';
import Home from './Home';

describe('Home component should', () => {
  afterEach(() => jest.clearAllMocks);
  const mockProps = {
    addItemHandler: jest.fn(),
    removeItemHandler: jest.fn(),
    categorisedProducts: {
      'MOCK CATEGORY 1': [{
        id: 1, name: 'MOCK 1', price: 10, itemCount: 2, count: 5, category: 'MOCK CATEGORY 1',
      }],
      'MOCK CATEGORY 2': [{
        id: 2, name: 'MOCK 1', price: 10, itemCount: 2, count: 5, category: 'MOCK CATEGORY 2s',
      }],
    },
  };
  it('should call two ProductList components', () => {
    render(<Home {...mockProps} />);
    screen.logTestingPlaygroundURL();
    const productListElement = screen.getAllByTestId('product-list');
    expect(productListElement.length).toBe(2);
  });
  it('should match the snapshot', () => {
    const { container } = render(<Home {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
