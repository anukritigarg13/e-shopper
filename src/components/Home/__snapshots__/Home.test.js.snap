// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should should match the snapshot 1`] = `
<div>
  <div
    class="product-container"
    data-testid="product-list"
  >
    <div
      class="product-type"
    >
      MOCK CATEGORY 1
    </div>
    <div
      class="product-list"
    >
      <div
        class="product"
        data-testid="product-test"
      >
        <div
          class="product-image-div"
        >
          <img
            alt="image of grapes"
            class="product-image"
            data-testid="product-img"
            src="assets/images/grapes.jpeg"
          />
        </div>
        <p>
          Fresco
        </p>
        <p>
          MOCK 1
        </p>
        <p>
          1 kg
        </p>
        <div
          class="quantity-placard"
        >
          <p>
            MRP
             
            10
            /-
          </p>
          <div
            class="quantity-control"
            data-testid="quantity-control"
          >
            <p>
              <button
                class="control-button"
                type="button"
              >
                +
              </button>
            </p>
            <p
              class="quantity-in-basket"
            >
              2
               
              in Basket
            </p>
            <p>
              <button
                class="control-button"
                type="button"
              >
                -
              </button>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="product-container"
    data-testid="product-list"
  >
    <div
      class="product-type"
    >
      MOCK CATEGORY 2
    </div>
    <div
      class="product-list"
    >
      <div
        class="product"
        data-testid="product-test"
      >
        <div
          class="product-image-div"
        >
          <img
            alt="image of grapes"
            class="product-image"
            data-testid="product-img"
            src="assets/images/grapes.jpeg"
          />
        </div>
        <p>
          Fresco
        </p>
        <p>
          MOCK 1
        </p>
        <p>
          1 kg
        </p>
        <div
          class="quantity-placard"
        >
          <p>
            MRP
             
            10
            /-
          </p>
          <div
            class="quantity-control"
            data-testid="quantity-control"
          >
            <p>
              <button
                class="control-button"
                type="button"
              >
                +
              </button>
            </p>
            <p
              class="quantity-in-basket"
            >
              2
               
              in Basket
            </p>
            <p>
              <button
                class="control-button"
                type="button"
              >
                -
              </button>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
